plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -626,28 +626,43 @@ public static KVector toRelative(final KVector point, final ElkNode parent) {
     }
 
     /**
-     * Get the edge section points as an array of vectors.
+     * Get the edge section points as an array of vectors. 
+     * Unnecessary bend points are filtered out automatically. 
      * 
      * @param section an edge section
-     * @return an array with all edge section points
+     * @return an array with all needed edge section points
      */
-    private static KVector[] getPoints(final ElkEdgeSection section) {
+    private static ArrayList<KVector> getPoints(final ElkEdgeSection section) {
         int n = section.getBendPoints().size() + 2;
-        KVector[] points = new KVector[n];
+        ArrayList<KVector> points = new ArrayList<KVector>(n);
 
         // Source point
-        points[0] = new KVector(section.getStartX(), section.getStartY());
+        points.add(new KVector(section.getStartX(), section.getStartY()));
 
         // Bend points
-        ListIterator<ElkBendPoint> pointIter = section.getBendPoints().listIterator();
-        while (pointIter.hasNext()) {
-            ElkBendPoint bendPoint = pointIter.next();
-            points[pointIter.nextIndex()] = new KVector(bendPoint.getX(), bendPoint.getY());
-        }
+        section.getBendPoints().stream()
+                .forEach(bendPoint -> points.add(new KVector(bendPoint.getX(), bendPoint.getY())));
 
         // Target point
-        points[n - 1] = new KVector(section.getEndX(), section.getEndY());
-
+        points.add(new KVector(section.getEndX(), section.getEndY()));
+
+        // Filter unnecessary bend points from the list
+        int i = 1;
+        while (i < points.size() - 1) {
+            // Unnecessary bend points are given if three points in a row have the same x or y coordinate
+            KVector p1 = points.get(i - 1);
+            KVector p2 = points.get(i);
+            KVector p3 = points.get(i + 1);
+            
+            if ((p1.x == p2.x && p2.x == p3.x) || (p1.y == p2.y && p2.y == p3.y)) {
```

Comment:
Equality on doubles may be problematic.

Reply:
Sure, but
a) we are talking about fixed layouter, meaning the coordinates are passed by the model and not some calculation that is likely to create small deltas
b) if there is an offset the edge isn't entirely straight, is it?